File: 4 Median of Sorted Array/MedianSolution.py
def findMedianSortedArrays(nums1, nums2):
    mergedArray = [0] * (len(nums1)+len(nums2))
    i,j,k=0,0,0
    while i<len(nums1) and j<len(nums2):
        if nums1[i]<nums2[j]:
            mergedArray[k]=nums1[i]
            i+=1
        else:
            mergedArray[k]=nums2[j]
            j+=1
        k+=1

    while i<len(nums1):
        mergedArray[k]=nums1[i]
        i+=1
        k+=1
    
    while j<len(nums2):
        mergedArray[k]=nums2[j]
        j+=1
        k+=1

    if len(mergedArray)%2==1:
        return float(mergedArray[len(mergedArray)/2])
    else:
        return float((mergedArray[len(mergedArray)/2]+mergedArray[len(mergedArray)-1]/2))

File: 4 Median of Sorted Array/MedianSolution.java
public class MedianSolution {
  public double findMedianSortedArrays(int[] nums1, int[] nums2) {
    int[] mergedNums = merge(nums1, nums2);
    if(mergedNums.length % 2 ==1){
        return mergedNums[mergedNums.length /2];
    }
    return ((double)mergedNums[mergedNums.length/2] + (double)mergedNums[(mergedNums.length/2) -1]) /2;
  }
  public int[] merge(int[] nums1, int[] nums2) {
    int[] mergedNums = new int[nums1.length + nums2.length];
    
    int i = 0, j = 0, k = 0;
    while (i < nums1.length && j < nums2.length) {
        if (nums1[i] < nums2[j]) {
            mergedNums[k] = nums1[i];
            i++;
        } else {
            mergedNums[k] = nums2[j];
            j++;
        }
        k++;
    }

    while (i < nums1.length) {
        mergedNums[k] = nums1[i];
        i++;
        k++;
    }

    while (j < nums2.length) {
        mergedNums[k] = nums2[j];
        j++;
        k++;
    }

    return mergedNums;
}

}
